src/test/java/org/folio/rmapi/builder/VendorUrlBuilderTest.java

```diff
@@ -0,0 +1,35 @@
+package org.folio.rmapi.builder;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class VendorUrlBuilderTest {
+  @Test
+  public void shouldBuildUrlForNameSortWhenSortName(){
+    String path = new VendorUrlBuilder().sort("name").build();
+    assertEquals("vendors?offset=1&count=25&orderby=VendorName", path);
+  }
+  @Test
+  public void shouldBuildUrlForRelevanceSortWhenSortRelevance(){
+    String path = new VendorUrlBuilder().sort("relevance").build();
+    assertEquals("vendors?offset=1&count=25&orderby=Relevance", path);
+  }
+
+  @Test
+  public void shouldBuildUrlForNameSortWhenSortIsNotSet(){
+    String path = new VendorUrlBuilder().build();
+    assertEquals("vendors?offset=1&count=25&orderby=VendorName", path);
+  }
+
+  @Test
+  public void shouldBuildUrlForRelevanceSortWhenSortIsNotSetAndQueryIsSet(){
+    String path = new VendorUrlBuilder().q("higher education").build();
+    assertEquals("vendors?search=higher+education&offset=1&count=25&orderby=Relevance", path);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void shouldThrowExceptionIfSortValueIsInvalid() {
+    new VendorUrlBuilder().sort("abc").build();
```

Comment:
Should we check for specific status code and error message in Rest Assured tests for this scenario?

Reply:
Invalid sort parameter will result in 500 error,
I moved validation of sort parameter to EholdingsProvidersImpl so that 400 error is returned.